### PR TITLE
Tests for expression transformation, parser code update

### PR DESCRIPTION
**Pull request by vratsel**

### CODE REVIEW

File: testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+public class AbstractParserTestCase
+{
+   /**
+    * The main test for all expressions. Asserts that expression should be transformed 
+    * correctly into expected result
+    * 
+    * @param expression to transform
+    * @param expected result
+    */
+   private void exressionTransformTest(String expression, String expected)
```
Comment (jesperpedersen) [inline testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java]: exressionTransformTest -> expressionTransformTest

File: testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+      exressionTransformTest("a${:}_${/}b", "a" + File.pathSeparator + "_" + File.separator + "b");
+   }
+
+   /**
+    * Test nested expressions
+    */
+   @Test
+   public void testNestedExpressions()
+   {
+      exressionTransformTest("${Prop1:${Prop2:default}1}", "default1");
```
Comment (jesperpedersen) [inline testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java]: That is not a valid expression - it should be ${Prop1:${Prop2:default}} -> default

File: testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+   }
+
+   /**
+    * Test nested expressions with some properties set
+    */
+   @Test
+   public void testNestedExpressionsWithPropertiesSet1()
+   {
+      System.setProperty("Prop3", "3");
+      exressionTransformTest("${Prop3:${Prop2:default}1}", "3");
```
Comment (jesperpedersen) [inline testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java]: Same deal here

File: testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+   }
+
+   /**
+    * Test nested expressions with some properties set
+    */
+   @Test
+   public void testNestedExpressionsWithPropertiesSet2()
+   {
+      System.setProperty("Prop4", "4");
+      exressionTransformTest("${Prop3:${Prop4:default}1}", "41");
```
Comment (jesperpedersen) [inline testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java]: And here

**Comment by vratsel** [comment]
Updated the code

**Comment by jesperpedersen** [comment]
Merged

**Pull request merged**